src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -64,41 +67,78 @@ const DescriptionRow = ({ onValueChanged, description }) => {
+    // coded value.
+
+    for (const s of vm.snapshots.sort((a, b) => b.creationTime - a.creationTime)) {
+        if (s.memoryPath)
+            return dirname(s.memoryPath);
+    }
+
     const devices = getSortedBootOrderDevices(vm).filter(d => d.bootOrder &&
                                                               d.device.device === "disk" &&
                                                               d.device.type === "file" &&
                                                               d.device.source.file);
     if (devices.length > 0) {
-        const primaryDiskPath = devices[0].device.source.file;
-        const directory = primaryDiskPath.substring(0, primaryDiskPath.lastIndexOf("/") + 1);
-        return directory + snapName;
+        return dirname(devices[0].device.source.file);
     } else {
         if (vm.connectionName === LIBVIRT_SYSTEM_CONNECTION)
-            return "/var/lib/libvirt/memory/" + snapName;
+            return "/var/lib/libvirt/memory";
         else if (current_user)
-            return current_user.home + "/.local/share/libvirt/memory/" + snapName;
+            return current_user.home + "/.local/share/libvirt/memory";
     }
 
     return "";
 }
 
-const MemoryPathRow = ({ onValueChanged, memoryPath, validationError }) => {
+const MemoryLocationRow = ({ onValueChanged, memoryLocation, validationError, available, needed }) => {
+    let info = "";
+    let info_variant = "default";
+
+    if (available) {
+        info = cockpit.format(_("$0 available at this location."), cockpit.format_bytes(available));
+        if (needed && available * 0.9 < needed)
+            info_variant = "warning";
+    }
+    if (needed) {
+        info = info + " " + cockpit.format(_("About $0 are needed to save the memory state of the machine."),
+                                           cockpit.format_bytes(needed));
+    }
+
     return (
-        <FormGroup id="snapshot-create-dialog-memory-path" label={_("Memory file")}>
+        <FormGroup id="snapshot-create-dialog-memory-location" label={_("Memory save location")}>
             <FileAutoComplete
-                onChange={value => onValueChanged("memoryPath", value)}
-                superuser="try"
+                onChange={value => onValueChanged("memoryLocation", value)}
+                value={memoryLocation}
                 isOptionCreatable
-                value={memoryPath} />
-            <FormHelper helperTextInvalid={validationError} />
+                onlyDirectories
+                placeholder={_("Path to directory")}
+                superuser="try" />
+            <FormHelper helperTextInvalid={validationError}
+                        helperText={info}
+                        variant={validationError ? "error" : info_variant} />
         </FormGroup>
     );
 };
 
+function get_available_space(path, callback) {
+    if (!path)
+        callback(null);
+
+    cockpit.script(get_available_space_sh, [path], { superuser: "try" })
```

Comment:
Might be worth conditionalizing the "try" on the system connection, and drop it (or set to False) for the session one?

Reply:
Yes, good point.